src/cmap/auth/mongodb_oidc.ts

```diff
@@ -95,19 +98,32 @@ export class MongoDBOIDC extends AuthProvider {
     handshakeDoc: HandshakeDocument,
     authContext: AuthContext
   ): Promise<HandshakeDocument> {
-    const { credentials } = authContext;
-
-    if (!credentials) {
-      throw new MongoMissingCredentialsError('AuthContext must provide credentials.');
+    const credentials = getCredentials(authContext);
+    const { connection } = authContext;
+    const mechanismProperties = credentials.mechanismProperties;
+    const allowedHosts = mechanismProperties.ALLOWED_HOSTS || DEFAULT_ALLOWED_HOSTS;
+    const providerName = mechanismProperties.PROVIDER_NAME;
+    // Cloud provider automatic auth does not validate allowed hosts.
```

Comment:
This seems reasonable but I’m not quite seeing where this behavior is written down in the spec – it should probably be mentioned there?

Reply:
I don't think the behaviour was explicitly called out in the spec but there is a prose test for it: https://github.com/mongodb/specifications/pull/1381/files#diff-93f94aaea2345c5e72ced4859fe1cefea928499f5b4019961c568256dbf72b62R136